Minor fixes for Issues (#1173)
hhvm/user-documentation: guides/hack/13-contexts-and-capabilities/02-local-operations.md

* `write_prop`, representing the permission to mutate properties of objects
* `dynamic`, representing the permission to cast a value to the `dynamic` type

> In all of the below cases, the relevant local operations are only legal due to the existence of the matching capabilities within the context of the function.

Reviewer: This is definitely an improvement too, but perhaps we should avoid 'legal'? How about this?

```suggestion
In all of the below cases, the relevant local operations typecheck due to the matching capabilities within the context of the function.
```
